### PR TITLE
Upgrade pg gems

### PR DESCRIPTION
**Pull request by javitonino**
Edit major gems to versions that are supported in Ruby 2.4. Should get rid of deprecation messages.

Major changes:
- Upgrade Sequel gem. Since both ActiveRecord and Sequel depend on the `pg` gem, we cannot update to the latest Sequel (5.x). So we must stay in 4.x. Inside 4.x, I choose a version that is Ruby 2.4 compatible and does not deprecate many things to minimize the PR (I promise I tried).
  - Update `sequel_pg`, `pg` and `sequel-rails` to compatible versions with this upgrade
  - From Sequel 3 to 4 there are some changes:
    - Methods like `"mitabla".lit` are now written like `Sequel.lit("mitabla")`. They do not add things to the string class anymore, yay :raised_hands:. **This accounts for most of the changed files** (migrations)
    - Some plugins are no longer built in (pagination, after_initialize) and have to be added
    - Sequel 4 has better error handling and some tests that should have been broken the whole time started failing now (Sequel 3 silently ignored some errors). Fixed.
    - More, I left comments in the code.
- Upgrade Redis gem. Just went to the latest that is supported by our current Resque gem. No changes needed.
- Removed old `event_machine` gem that depended on `pg` and wasn't used anymore. :recycle: 
- Ruby 2.4 adds a deprecation warning to `delegate/forwardable` if the called method is private. So I had to make some private methods public.

### CODE REVIEW

File: app/models/carto/user_service.rb [hunk 1]
```diff
@@ -148,8 +148,6 @@ def organization_member_group_role_member_name
       ).first['org_member_role']
     end
 
-    private
```
Comment (javitonino) [inline app/models/carto/user_service.rb]: These methods are made public using forwardable from the user model. That is, in the user model there is:
```
  delegate [
    :database_username, :database_password, :in_database,
[...]
  ] => :service
```

This exposes private methods of `user_service` and makes them public in `user` and Ruby 2.4 emits a deprecated warning. The solution is to make them public in the service.

File: app/models/carto/user_table.rb [hunk 2]
```diff
@@ -114,6 +114,10 @@ def dependent_visualizations
       affected_visualizations.select { |v| v.dependent_on?(self) }
     end
 
+    def affected_visualizations
```
Comment (javitonino) [inline app/models/carto/user_table.rb]: Making this public, same as above.

File: app/models/map.rb [hunk 3]
```diff
@@ -169,20 +169,12 @@ def self.provider_for_baselayer_kind(kind)
 
   # (lat,lon) points on all map data
   def center_data
-    (center.nil? || center == '') ? DEFAULT_OPTIONS[:center] : center.gsub(/\[|\]|\s*/, '').split(',')
+    center.presence || DEFAULT_OPTIONS[:center]
```
Comment (javitonino) [inline app/models/map.rb]: In Sequel 3, `center` was a string like `"[12,22]"`. Now, it is an actual array, so code can be much simpler.

File: app/models/map.rb [hunk 4]
```diff
@@ -169,20 +169,12 @@ def self.provider_for_baselayer_kind(kind)
...
-    else
-      bbox_sw = view_bounds_sw.gsub(/\[|\]|\s*/, '').split(',').map(&:to_f)
-    end
-    if view_bounds_ne.nil? || view_bounds_ne == ''
-      bbox_ne = DEFAULT_OPTIONS[:bounding_box_ne]
-    else
-      bbox_ne = view_bounds_ne.gsub(/\[|\]|\s*/, '').split(',').map(&:to_f)
-    end
+    bbox_sw = view_bounds_sw.presence || DEFAULT_OPTIONS[:bounding_box_sw]
+    bbox_ne = view_bounds_ne.presence || DEFAULT_OPTIONS[:bounding_box_ne]
```
Comment (javitonino) [inline app/models/map.rb]: Same as `center` above

File: services/importer/lib/importer/column.rb [hunk 5]
```diff
@@ -38,7 +38,7 @@ class Column
       def initialize(db, table_name, column_name, user, schema = DEFAULT_SCHEMA, job = nil, logger = nil, capture_exceptions = true)
         @job          = job || Job.new({logger: logger})
         @db           = db
-        @table_name   = table_name.to_sym
+        @table_name   = table_name
```
Comment (javitonino) [inline services/importer/lib/importer/column.rb]: This is passed to `db.schema` which is described to take an string. Sequel 3 was more lenient allowing symbols as well, Sequel 4 is more strict.

File: spec/models/carto/user_creation_spec.rb [hunk 6]
```diff
@@ -53,6 +53,11 @@
   describe 'validation token' do
     include_context 'organization with users helper'
 
+    before(:each) do
+      # Disable central for these tests
+      Cartodb::Central.stubs(:sync_data_with_cartodb_central?).returns(false)
```
Comment (javitonino) [inline spec/models/carto/user_creation_spec.rb]: During using creation, the hook is called to sync data with central. Sequel 3 was not properly handling this error and it silently ignored it. The upgrade to Sequel 4 makes this error raise an exception and the tests to fail. They should have been failing the whole time. We just disable central (like other tests in this suite).

File: spec/models/organization_spec.rb [hunk 7]
```diff
@@ -65,7 +65,7 @@ def get_organization
     end
 
     it 'Destroys users and owner as well' do
-      organization = Organization.new(quota_in_bytes: 1234567890, name: 'wadus', seats: 5).save
+      organization = Organization.new(quota_in_bytes: 123456789000, name: 'wadus', seats: 5).save
```
Comment (javitonino) [inline spec/models/organization_spec.rb]: Same as above, Sequel 4 has better error handling and we were actually going over quota when creating users but Sequel was swallowing the errors.

File: spec/requests/carto/api/visualizations_controller_spec.rb [hunk 8]
```diff
@@ -21,6 +21,10 @@
   include VisualizationDestructionHelper
   include FeatureFlagHelper
 
+  before(:all) do
+    create_account_type_fg('ORGANIZATION USER')
```
Comment (javitonino) [inline spec/requests/carto/api/visualizations_controller_spec.rb]: Same as above, Sequel 3 ignoring errors.

File: spec/requests/sessions_controller_spec.rb [hunk 9]
```diff
@@ -535,6 +535,7 @@ def create_admin_user(organization)
 
       before(:all) do
         create
+        Cartodb::Central.stubs(:sync_data_with_cartodb_central?).returns(false)
```
Comment (javitonino) [inline spec/requests/sessions_controller_spec.rb]: Same as above, Sequel 3 ignoring errors.

File: app/models/carto/search_tweet.rb [hunk 10]
```diff
@@ -13,7 +13,7 @@ def self.twitter_imports_count(query, date_from, date_to)
       query
         .where('search_tweets.state' => ::SearchTweet::STATE_COMPLETE)
         .where('search_tweets.created_at >= ? AND search_tweets.created_at <= ?', date_from, date_to + 1.days)
-        .sum("search_tweets.retrieved_items".lit).to_i
+        .sum("search_tweets.retrieved_items").to_i
```
Comment (gonzaloriestra) [inline app/models/carto/search_tweet.rb]: No need to use `Sequel.lit` here?

Comment (javitonino) [reply app/models/carto/search_tweet.rb]: No, because it's an ActiveRecord model.

**Pull request merged**